hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/tasks/FSONSSummaryTask.java

```diff
@@ -0,0 +1,216 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hadoop.ozone.recon.tasks;
+import org.apache.commons.lang3.tuple.ImmutablePair;
+import org.apache.commons.lang3.tuple.Pair;
+import org.apache.hadoop.hdds.utils.db.Table;
+import org.apache.hadoop.hdds.utils.db.TableIterator;
+import org.apache.hadoop.ozone.om.OMMetadataManager;
+import org.apache.hadoop.ozone.om.helpers.OmDirectoryInfo;
+import org.apache.hadoop.ozone.om.helpers.OmKeyInfo;
+import org.apache.hadoop.ozone.om.helpers.WithParentObjectId;
+import org.apache.hadoop.ozone.recon.api.types.NSSummary;
+import org.apache.hadoop.ozone.recon.spi.ReconNamespaceSummaryManager;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Inject;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+
+import static org.apache.hadoop.ozone.om.OmMetadataManagerImpl.DIRECTORY_TABLE;
+import static org.apache.hadoop.ozone.om.OmMetadataManagerImpl.FILE_TABLE;
+
+/**
+ * Class for handling FSO specific tasks.
+ */
+public class FSONSSummaryTask extends NSSummaryTask {
+
+  private static final Logger LOG =
+      LoggerFactory.getLogger(FSONSSummaryTask.class);
+
+  @Inject
+  public FSONSSummaryTask(ReconNamespaceSummaryManager
+                            reconNamespaceSummaryManager) {
+    super(reconNamespaceSummaryManager);
+  }
+
+  @Override
+  public String getTaskName() {
+    return "FSONSSummaryTask";
+  }
+
+  // We only listen to updates from FSO-enabled KeyTable(FileTable) and DirTable
+  public Collection<String> getTaskTables() {
+    return Arrays.asList(new String[]{FILE_TABLE, DIRECTORY_TABLE});
```

Comment:
nit: redundant `new String[]`
```suggestion
    return Arrays.asList(FILE_TABLE, DIRECTORY_TABLE);
```